test/git_checkout_test.go

```diff
@@ -0,0 +1,185 @@
+// +build e2e
+
+/*
+Copyright 2018 Knative Authors LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+	tb "github.com/knative/build-pipeline/test/builder"
+	knativetest "github.com/knative/pkg/test"
+	"github.com/knative/pkg/test/logging"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	gitSourceResourceName  = "git-source-resource"
+	gitTestTaskName        = "git-check-task"
+	gitTestPipelineName    = "git-check-pipeline"
+	gitTestPipelineRunName = "git-check-pipeline-run"
+)
+
+// TestGitPipelineRun is an integration test that will verify the source code is either fetched or pulled
+// successfully under different revision inputs (branch, commitid, tag, ref)
+func TestGitPipelineRun(t *testing.T) {
+	logger := logging.GetContextLogger(t.Name())
+	t.Parallel()
+
+	revisions := []string{"c15aced0e5aaee6456fbe6f7a7e95e0b5b3b2b2f", "c15aced", "release-0.1", "v0.1.0", "refs/pull/347/head"}
```

Comment:
can we add a case for `master` to make sure everything is fine when its using the fetch

Reply:
Done.